IPython/core/tests/test_history.py

```diff
@@ -168,6 +169,28 @@ def test_magic_rerun():
     ip.run_cell("%rerun", store_history=True)
     nt.assert_equal(ip.user_ns["a"], 12)
 
+def test_magic_rerun_search():
+    """Test for %rerun -g (searching for text match)"""
+    ip = get_ipython()
+    ip.run_cell("a = 10", store_history=True)
+    ip.run_cell("a += 1", store_history=True)
+    nt.assert_equal(ip.user_ns["a"], 11)
+    ip.run_cell("%rerun -g 10", store_history=True)
+    nt.assert_equal(ip.user_ns["a"], 10)
+
+
+def test_magic_rerun_quiet():
+    """Simple test for %rerun -q (quiet, no echoing of output)"""
+    ip = get_ipython()
+    ip.run_cell("a = 10", store_history=True)
+    ip.run_cell("a += 1", store_history=True)
+    nt.assert_equal(ip.user_ns["a"], 11)
+    with capture_output() as cap:
+        ip.run_cell("%rerun -q", store_history=True)
+    nt.assert_equal(ip.user_ns["a"], 12)
+    nt.assert_not_in("a += 1", cap.stdout)
```

Comment:
Nice work on the tests!

Reply:
Thanks, nice work on the support functions like `capture_output`! ;)